test/easyconfigs/easyconfigs.py

```diff
@@ -186,12 +186,19 @@ def check_dep_vars(dep, dep_vars):
             if dep == 'binutils':
                 empty_vsuff_vars = [v for v in dep_vars.keys() if v.endswith('versionsuffix: ')]
                 if len(empty_vsuff_vars) == 1:
-                    dep_vars = [(k, v) for (k, v) in dep_vars.items() if k != empty_vsuff_vars[0]]
+                    dep_vars = dict((k, v) for (k, v) in dep_vars.items() if k != empty_vsuff_vars[0])
             # filter out FFTW and imkl with -serial versionsuffix which are used in non-MPI subtoolchains
             elif dep in ['FFTW', 'imkl']:
                 serial_vsuff_vars = [v for v in dep_vars.keys() if v.endswith('versionsuffix: -serial')]
                 if len(serial_vsuff_vars) == 1:
-                    dep_vars = [(k, v) for (k, v) in dep_vars.items() if k != serial_vsuff_vars[0]]
+                    dep_vars = dict((k, v) for (k, v) in dep_vars.items() if k != serial_vsuff_vars[0])
+
+            # filter out variants that are specific to a particular version of CUDA
+            cuda_dep_vars = [v for v in dep_vars.keys() if '-CUDA' in v]
+            if len(dep_vars) > len(cuda_dep_vars):
```

Comment:
what's the purpose of this `if`?

Reply:
If we only have easyconfigs with `-CUDA-` in the `versionsuffix`, then we shouldn't be filtering them out at all, otherwise `dep_vars` ends up empty, and then the stuff below fails...

Example: `magma` (only `2018a` easyconfig is `magma-2.3.0-foss-2018a-CUDA-9.1.85.eb`)